main.cpp

```diff
@@ -285,6 +284,23 @@ struct FloatType
     FloatType& pow(const DoubleType&);
     FloatType& pow(float);
 
+    FloatType& apply(std::function<FloatType& (float&)> FloatFptr)
+    {
+        if (FloatFptr)
+        {
+            return FloatFptr(*value);
+        }
+        return *this;
+    }
+
+    FloatType& apply(void(*FloatFptr)(float&))
```

Comment:
Follow the coding standard in the course.
Variable names need to start with a lowercase letter.
`FloatFptr` needs to be renamed.
- [ ] revise